Realistic phase1 hf reco

igv4321 opened the pull request:
Implementation of the "HFFlexibleTimeCheck" Phase 1 HF reconstruction algorithm with associated database classes. Some refactoring of the existing code to eliminate duplication.

This pull request has been discussed at
https://indico.cern.ch/event/559519/contributions/2257746/attachments/1317838/1975047/notes.pdf

I would appreciate quick approval. There are some changes in DataFormats needed for subsequent developments.

makortel committed:
Add possibility to weigh pixel hits in QuickTrackAssociatorByHits

Use weighted number of hits also in the denominator

Commit:
extendedO2O

makortel committed:
Bugfixes in SeedingLayerSetsHits

Spudmeister committed:
Expanded debuggin information to try and understand the location of the
segfault

dtnrm committed:
Disable web/http protocol for curl storage plugin

dtnrm committed:
New DavixAdaptor plugin skeleton
Support http/web protocols and implemented basic functions: open, read, position

dtnrm committed:
Davix close/abort function

dtnrm committed:
Implement Davix Vector read operation

dtnrm committed:
Implement Davix Vector read operation with offset option

dtnrm committed:
Authentication priority check:
  1. X509_USER_PROXY
  2. /tmp/x509up_u<(geteuid)>
  3. X509_USER_{CERT|KEY}
If none of these is present, it will try to make connection without proxy/cert. X509_CERT_DIR can also be overwritten with environment variable.

dtnrm committed:
Davix_Debug is controlled with flags [0-4] through env variable:
  0. Default davix log level;
  1. DAVIX_LOG_WARNING;
  2. DAVIX_LOG_VERBOSE;
  3. DAVIX_LOG_DEBUG;
  4. DAVIX_LOG_ALL;
If provided value is not integer or it is not able to parse to int, it will use 0. Any other number from 4toN will be considered as 4. DAVIX_LOG_ALL which is most verbose;

Spudmeister committed:
Fixed segfault issue.  Still putting nullptr or unique ptr in event

dtnrm committed:
Improve logging for DavixAdaptor plugin in all functions
It checks DavixErr if it was set or not also if FD was opened sucessfully and if return value is greater than 0. 0 - means it is end of file. If any of these are found it will raise an issue.

dtnrm committed:
Close Davix_FD before opening new one

Martin-Grunewald committed:
Toward July6th train

Andrej-CMS committed:
filter

Martin-Grunewald committed:
HLT development 1

Spudmeister committed:
Seems to be unpacking now

Spudmeister committed:
Switched printout to hex

Martin-Grunewald committed:
Move to L1T v5 and resulting seed changes

Martin-Grunewald committed:
Update TSG MC test file

dtnrm committed:
Add missing BuildFile for Davix plugin

dtnrm committed:
DavixAdaptor unique_ptrs, global var, null fixes
Do not use global variable for Davix Context
Use unique_ptr in places where it is possible;
Define m_name before throwing an error, otherwise it is null
Check variables without NULL.

dtnrm committed:
Update Davix exceptions, braces, compare return bytes
Throw exception if open is called on an open file
Compare bytes requested vs read and throw exceptions
Surround with braces {} as done in the rest of the code

Andrej-CMS committed:
ttHFGenFilter

Martin-Grunewald committed:
Closing in on HLT menu v3.0

Martin-Grunewald committed:
More EOS files

dtnrm committed:
DavixError has to be NULL and we can't check vector read return
1)During initialization if we want to check and get DavixError it has to be NULL;
2) Currently Davix limits number of requests sent to the server in the interest of improving performance. This means that we can`t rely on value returned by preadVec if buffers > 1. It would work only in few cases, whenever ranges are not coalescing. If two ranges are overlapping by a single byte, it will contain one byte less in return.

Martin-Grunewald committed:
Bump path version numbers wrt V383

Spudmeister committed:
Updated based on new specs

Spudmeister committed:
Updating based on new specs

mandrenguyen committed:
Point to 50ns vtx smearing for boosted pPb and clean up

schneiml committed:
Implement EXTEND_X on 1D histos

schneiml committed:
Redo Configuration, Centralize more

schneiml committed:
Implement TProfiles in step1.

schneiml committed:
Turn on profiles in config, other fixes (PerModule)

dtnrm committed:
Add unit tests for Davix plugin
This pull request makes sure that old test which are on StorageFactory/test will work as it should be.
Also new tests are checking if file is accessible, it can do read, vector read with offset and also check if filenotfound or wrong url will raise proper errors.

schneiml committed:
Put the quantitiy in the title, such that we don't loose it in 2D profiles.

dtnrm committed:
Move http tests from Utilities/StorageFactory to Utilities/DavixAdaptor

makortel committed:
Replace TMatrix with Eigen in FastCircleFit, do all calculations in float

With double precision there would be no changes. Moving at the same to
float incurs numerical differences, but in simple tests those seem to
be smaller than 1 % in the circle parameters. I think this is
acceptable given that the class is called "Fast...".

makortel committed:
Fully inline FastCircleFit

Needed for generic interface, especially for avoiding DynArray when
std::array's are given, i.e. we know statically the size.

makortel committed:
Add DynArray::data()

makortel committed:
Restructure RZLine calculations, generalize construction, inline

makortel committed:
Move to std::array, avoid sqr(sqrt())

makortel committed:
Move RZLine.h to interface

Since it's already #included outside this package, interface is the
proper place (especially now that the class is header-only)

schneiml committed:
More confguration improvements

Commit:
extendedO2O for boost->std change in 81x

Commit:
extendedO2O for boost->std change in 81x

uhussain committed:
Fixed static analyzer warnings in this file

uhussain committed:
Fixed static analyzer warnings in HcalUnpacker

gouskos committed:
dqm plot, pixel endcap roc occupancy in 81x

uhussain committed:
Deleted redundant cast in line 85

perrotta committed:
Update HLT menu up to /dev/CMSSW_8_0_0/HLT/V433, and new L1 menu v6

uhussain committed:
Manually undid the incorrect and unneeded changes

Commit:
L1Menu-only L1T O2O for 81x

Commit:
L1Menu-only L1T O2O for 81x

Commit:
put back slow unit tests

Commit:
put back slow unit tests

Commit:
addressing most of the memory leaks

Commit:
addressing most of the memory leaks

Commit:
merging for rebasing

schneiml committed:
Fix TProfile trend plots.

The Counting+Trend case is still not cleanly solved, but usable now.

Commit:
Removing AbsHFPhase1AlgoData.h and adding AbsHcalAlgoData.h instead

Commit:
Switching algos to use AbsHcalAlgoData for configuration

Commit:
classes to configure Phase 1 Hcal reco algorithms

Commit:
adding HFPhase1PMTParams db read/write modules

Commit:
added detector id transforms for lookups

Commit:
Add AlCaReco and DQM code for HcalIsolatedBunch

makortel committed:
Document constructor variants

perrotta committed:
Update HLT menu up to /dev/CMSSW_8_0_0/HLT/V438

Spudmeister committed:
Cleanup

uhussain committed:
Fixing errors related to variables pData16 & begin32

kpedro88 committed:
remove commented/unneeded code

Andrej-CMS committed:
ttHFGenFilter now checks if additional b-hadron comes from ISR. based on top mothers

Andrej-CMS committed:
Added commentary on the code

venturia committed:
New macro to produce trend plots of the cluster occupancy

Commit:
Add changes to standard sequences

perrotta committed:
Update HLT menu up to /dev/CMSSW_8_0_0/HLT/V440

Commit:
fix: eval_auto_bounds returns good value for pt==pt_low

Spudmeister committed:
Unique ptr update

Commit:
Change output part for HcalIsolatedBunchFilter

Commit:
Change AlCaRecoStreamsHeavyIons_cff to include IsolatedBunchSelector

mmusich committed:
Update Run-I GTs with FEMapRcd and Phase-I GTs with reconstruction Geometry

Commit:
Take care of Davids comments

Commit:
Check

Commit:
Check

perrotta committed:
Update HLT menu up to /dev/CMSSW_8_0_0/HLT/V448

Commit:
adding database reader/writer modules for HFPhase1PMTParams

Andrej-CMS committed:
added VarParsing to testConfig

perrotta committed:
Update HLT menu up to /dev/CMSSW_8_0_0/HLT/V450

uhussain committed:
Removed unnecessary spaces in L605,606,650,854

perrotta committed:
Update MC input file for tests

Commit:
adding HFFlexibleTimeCheck algorithm

Commit:
removing HcalPhase1AlgoDBModules

Commit:
added parser for HFFlexibleTimeCheck parameters

Spudmeister committed:
Pointer work

Spudmeister committed:
Pointer work

Commit:
get rid of unused variable modules

Commit:
various db-related mods

Spudmeister committed:
Pointer work

Spudmeister committed:
Pointer work

jan-kaspar committed:
Directories rename: Totem -> CTPPS.

jan-kaspar committed:
DataFormats/CTPPSDetId: paths upated

jan-kaspar committed:
DataFormats/CTPPSDigi: paths updated

jan-kaspar committed:
CondFormats/CTPPSReadoutObjects: paths updated

jasperlauwers committed:
Adding Mu12_Ele23 path

jan-kaspar committed:
EventFilter/CTPPSRawToDigi: paths updated

jan-kaspar committed:
Geometry/VeryForwardGeometryBuilder: paths updated

jan-kaspar committed:
RecoCTPPS: paths updated

jan-kaspar committed:
DQM/CTPPS: paths updated

kpedro88 committed:
add new pulse shape for HE 2017 SiPM

jan-kaspar committed:
DQM/Integration: paths updated

kpedro88 committed:
make mcShape and recoShape configurable parameters for HcalHardcodeCalibrations

jan-kaspar committed:
Configuration/StandardSequences: paths updated

jan-kaspar committed:
bugfix: removed reference to non-existing config

Commit:
fixing status bits

dtnrm committed:
Update Davix plugin (nullptr, use vector, style casts, override)
DavixError must always be nullptr;
Override to each virtual function for the base class;
Use CMS naming conventions and DavixRequestParams no need to be private. Only used during open.
Remove left debugging statements;
During abort/close check davixError and in case it exists, raise exception.
Use std::vector for DavIOVecInput & DavIOVecOuput
Do not use old C' style casts and do static_cast<>()...

VinInn committed:
update pixelCluster ro support PhaseII

VinInn committed:
fix misstype

Commit:
adding HcalChebyshevFunctor and other mods

Commit:
put a config class name

uhussain committed:
correcting unncessary spaces and changes

Commit:
Error propagation added.

VinInn committed:
wrong shift

kpedro88 committed:
remove HcalCholeskyMatrix object entirely

Commit:
renaming the SignalAsymmetry flag

Spudmeister committed:
Registered flag

kpedro88 committed:
remove cholesky record

kpedro88 committed:
remove HcalCovarianceMatrix object entirely

Commit:
adding HcalLinearCompositionFunctor class

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
A new Pull Request was created by  @igv4321 (Igor Volobouev) for CMSSW_8_1_X.

It involves the following packages:

CondCore/HcalPlugins
CondCore/Utilities
CondFormats/DataRecord
CondFormats/HcalObjects
CondTools/Hcal
DataFormats/HcalRecHit
DataFormats/METReco
RecoLocalCalo/HcalRecAlgos
RecoLocalCalo/HcalRecProducers


@ghellwig, @cvuosalo, @cerminar, @cmsbuild, @franzoni, @slava77, @ggovi, @mmusich, @davidlange6 can you please review it and eventually sign? Thanks.
@ghellwig, @apfeiffer1, @ahinzmann, @rappoccio, @mmarionncern, @TaiSakuma, @tocheng, @argiro, @jdolen, @nhanvtran, @schoef, @mariadalfonso this is something you requested to watch as well.
@slava77, @smuzaffar you are the release manager for this.

cms-bot commands are list here cms-sw/cmssw#13028

Commit:
dont produce GEMCoPadDigi in 2016

ferencek committed:
added possibility to produce subjet TagInfos using fat jets but without SV clustering

cmsbuild commented:
-1

Tested at: 59c05903434f992dd9fb0ed384cecae7528a705a

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-15324/14276/summary.html

I found follow errors while testing this PR

**Failed tests:  UnitTests**

* **Unit Tests**:

I found errors in the following unit tests: 
 
 ---> test runtestRecoEgammaElectronIdentification had ERRORS

Commit:
throw an exception in case of failure

cmsbuild commented:
Pull request #15324 was updated. @ghellwig, @cvuosalo, @cerminar, @cmsbuild, @franzoni, @slava77, @ggovi, @mmusich, @davidlange6 can you please check and sign again.

igv4321 commented:
The test failure is not related to this PR. Some root file is missing instead.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-15324/14276/summary.html

cmsbuild committed:
Merge pull request #15146 from uhussain/CMSSW_8_1_X

Fixed all static analyzer warnings in MatacqRawEvent.cc, MatacqRawEvent.h & HcalUnpacker.cc

davidlange6 committed:
Merge pull request #15260 from makortel/improveFastCircleRZLine

Switch FastCircleFit to use Eigen, generalize FastCircleFit and RZLine interfaces

dtnrm committed:
Make it thread safe calling davix_set_log_level

perrotta committed:
Update HLT menu up to /dev/CMSSW_8_0_0/HLT/V452

kpedro88 committed:
Fix hardcode pedestal

Commit:
Take David's comment

davidlange6 committed:
Merge pull request #15258 from bsunanda/Run2-alca51

bsunanda:Run2-alca51 Add AlCaReco and DQM code for HcalIsolatedBunch

perrotta committed:
Update HLT menu up to /dev/CMSSW_8_0_0/HLT/V453

cmsbuild committed:
Merge pull request #15130 from juztas/libdavix

Change current curl fork/exec with libdavix plugin for read operations

davidlange6 committed:
Merge pull request #15283 from HeinerTholen/btag-fix-auto_eval_bounds

fix: eval_auto_bounds returns good value for pt==pt_low

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

slava77 commented:
@igv4321 
please provide a test configuration (either as a cmsDriver or similar command or as a separate cfg in this PR)
Thank you.

igv4321 commented:
Hi Slava,

There is no way to test this by providing just a separate cfg,
you also need a private database, as pointed out in the slides
mentioned in the PR submission comments.

I can put it somewhere on lxplus and email you the location.
Will this work?

Igor

On 07/29/2016 11:11 AM, Slava Krutelyov wrote:
>
> @igv4321 <https://github.com/igv4321>
> please provide a test configuration (either as a cmsDriver or similar command or as a separate cfg in this PR)
> Thank you.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/cms-sw/cmssw/pull/15324#issuecomment-236223159>, or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AFS4-YMi52Zw3l54rWTgy5_Qk9c_1kVlks5qaiZIgaJpZM4JXio9>.
>

slava77 commented:
On 7/29/16 9:27 AM, Igor Volobouev wrote:
> Hi Slava,
> 
> There is no way to test this by providing just a separate cfg,
> you also need a private database, as pointed out in the slides
> mentioned in the PR submission comments.
> 
> I can put it somewhere on lxplus and email you the location.
> Will this work?
> 

Hi Igor

Yes, this should work.

Thank you.


> Igor
> 
> On 07/29/2016 11:11 AM, Slava Krutelyov wrote:
>>
>> @igv4321 <https://github.com/igv4321>
>> please provide a test configuration (either as a cmsDriver or similar command or as a separate cfg in this PR)
>> Thank you.
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub <https://github.com/cms-sw/cmssw/pull/15324#issuecomment-236223159>, or mute the thread 
>> <https://github.com/notifications/unsubscribe-auth/AFS4-YMi52Zw3l54rWTgy5_Qk9c_1kVlks5qaiZIgaJpZM4JXio9>.
>>
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/15324#issuecomment-236227177>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbhh-jbhtJylMnN2285S5N0WvqtXQks5qainfgaJpZM4JXio9>.
>

mmusich commented:
@igv4321 @slava77 the preparation conditions DB is meant for these testing purposes. We can upload it for you in case of need.

slava77 commented:
On 7/29/16 10:18 AM, Marco Musich wrote:
> @igv4321 <https://github.com/igv4321> @slava77
> <https://github.com/slava77> the preparation conditions DB is meant for
> these testing purposes. We can upload it for you in case of need.

This would be nice.

> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/15324#issuecomment-236239355>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbluZzx5bbe4QOvA8HCzDrP_Wrg6cks5qajXPgaJpZM4JXio9>.
>

cmsbuild commented:
-1

Tested at: f2734c7b29b226480f1efc887de88c8ae73f121f

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-15324/14285/summary.html

I found follow errors while testing this PR

**Failed tests:  UnitTests**

* **Unit Tests**:

I found errors in the following unit tests: 
 
 ---> test runtestRecoEgammaElectronIdentification had ERRORS

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-15324/14285/summary.html

igv4321 commented:
@slava77 @mmunich

You can find the database on lxplus:
~igv/public/HFPhase1PMTParams_V00_mc.db

It contains the object of type "HFPhase1PMTParams" with
tag "HFPhase1PMTParams_V00_mc"

The script to run in order to test the new code is
~igv/public/SinglePi_Phase1.sh

Copy it and run like this: "./SinglePi_Phase1.sh -m 3".
It assumes that the file "HFPhase1PMTParams_V00_mc.db"
can be found in the current directory.

I am not sure which modifications have to be made
in order to switch the usage to the test database.

On 07/29/2016 12:20 PM, Slava Krutelyov wrote:
> On 7/29/16 10:18 AM, Marco Musich wrote:
> > @igv4321 <https://github.com/igv4321> @slava77
> > <https://github.com/slava77> the preparation conditions DB is meant for
> > these testing purposes. We can upload it for you in case of need.
>
> This would be nice.
>
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/cms-sw/cmssw/pull/15324#issuecomment-236239355>, or
> > mute the thread
> > <https://github.com/notifications/unsubscribe-auth/AEdcbluZzx5bbe4QOvA8HCzDrP_Wrg6cks5qajXPgaJpZM4JXio9>.
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/cms-sw/cmssw/pull/15324#issuecomment-236239991>, or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AFS4-bjb_ycCNA22CdJ-8_wqY1H-qIIRks5qajZqgaJpZM4JXio9>.
>

ferencek committed:
added possibility to produce subjet TagInfos using fat jets but without intermediate groomed fat jets

ferencek committed:
minor adaptations to take advantage of the new features in the SVTagInfo producer

ferencek committed:
preserve reference to the original jet the input PAT jet was derived from

venturia committed:
New common mode analyzer

barvic committed:
81X - Added check to the CSC Unpacker for FED/DDU<->chamber mapping inconsistencies to prevent CSC reco crashes in case of rare data corruptions. Fixed few reported CMS static analyzer warnings.

venturia committed:
Description of the CommonModeAnalyzer code

venturia committed:
Update README.md

venturia committed:
Description of the TrendPlotSingleBin macro

venturia committed:
Update README.md

venturia committed:
added a comment in a line

uhussain committed:
Fixing variable names to abide by CMS coding rules

uhussain committed:
reco::PFBlockCollection needs to be auto_ptr so changed unique_ptr to auto_ptr back

mmusich committed:
update L1T menu to v6 in all Global Tags

mmusich commented:
@slava77 the payload is available in also available in prep DB:
https://cms-conddb.cern.ch/cmsDbBrowser/list/Prep/tags/HFPhase1PMTParams_V00_mc

connections string: `frontier://FrontierPrep/CMS_CONDITIONS`
tag name: `HFPhase1PMTParams_V00_mc`

abdoulline commented:
Igor,   

(just for the record)
as we've agreed at HCAL Phase 1 Software TF meeting, in the next PR(s), please consider adding some (publicly avaivable in CMSSW) functionality for simple dumping of HFPhase1PMTParams parameters. 

Existing ASCII handling of _regular_ HCAL conditions 
http://cmslxr.fnal.gov/source/CalibCalorimetry/HcalAlgos/src/HcalDbASCIIIO.cc

https://twiki.cern.ch/twiki/bin/viewauth/CMS/HCAL/HcalOfflineDBInstructions -> Hcal offline DB instructions for 2016 (8X) -> Dumping 

is very helpful and used all the time by many people for various purposes, including validation/sanity checks for AlCaDB.

Thank you.

abdoulline commented:
Igor,   

(just for the record)
as we've agreed at HCAL Phase 1 Software TF meeting, in the next PR(s), please consider adding some (publicly avaivable in CMSSW) functionality for simple dumping of HFPhase1PMTParams parameters. 
Existing ASCII handling of _regular_ HCAL conditions 
http://cmslxr.fnal.gov/source/CalibCalorimetry/HcalAlgos/src/HcalDbASCIIIO.cc

https://twiki.cern.ch/twiki/bin/viewauth/CMS/HCAL/HcalOfflineDBInstructions -> Hcal offline DB instructions for 2016 (8X) -> Dumping 

is very helpful and used all the time by many people for various purposes, including validation/sanity checks for AlCaDB approvals.

Thank you.

perrotta committed:
Merged 81XHLTJuly29thTrain from repository cms-tsg-storm

perrotta committed:
Import commit 043465e892d85c290e48a96d38d4d6bd7d9de458 from mmusich

perrotta committed:
Merged 81XHLTJuly29thTrain from repository cms-tsg-storm

perrotta committed:
Revert "Import commit 043465e892d85c290e48a96d38d4d6bd7d9de458 from mmusich"

This reverts commit 7e0573882ad2718ecca040c601e7781a8ba86c26.

perrotta committed:
Merged 81XHLTJuly29thTrain from repository cms-tsg-storm

fabozzi committed:
new BPH MC relvals

perrotta committed:
Same as commit 043465e892d85c290e48a96d38d4d6bd7d9de458 from mmusich

perrotta committed:
Revert "Same as commit 043465e892d85c290e48a96d38d4d6bd7d9de458 from mmusich"

This reverts commit a5f1131f4bfca4c60cda4a3b09d5a3b6b4b038eb.

perrotta committed:
Merged 81XHLTJuly29thTrain from repository cms-tsg-storm

perrotta committed:
Solve conflicts

Commit:
Fix http proxy setting in curl for Tier0 DataService query in DQM/Integration/python/config/FrontierCondition_GT_autoExpress_cfi.py.

dntaylor committed:
remove SET from constructed

boudoul committed:
fixing typo in geom

boudoul committed:
typo...

fioriNTU committed:
IsolatedBunchesMonitor81X

cmsbuild committed:
Merge pull request #15336 from fabozzi/from-CMSSW_8_1_X_newrelvals

new BPH MC relvals

cmsbuild committed:
Merge pull request #15327 from cms-tsg-storm/81XHLTJuly29thTrain

81X HLT July 29th train

uhussain committed:
auto_ptr successfully changed to unique_ptr

ianna committed:
Add CutTubs DD Shape

ianna committed:
Make use of a new DD CutTubs shape

ianna committed:
Register the shape. Add comments. Bug fix.

ianna committed:
Unit test

jalimena committed:
use getByToken and other 81X updates

jalimena committed:
changed debug bools to LogDebug

Andrej-CMS committed:
added new flag and commentary

Commit:
Correct G4 validation code using isolated particles

cmsbuild committed:
Merge pull request #15330 from barvic/csc_unpacker_fix_for_mapping_inconsistencies_81X

81X - Added CSC Unpacker check for FED/DDU<->chamber mapping inconsistencies

cmsbuild committed:
Merge pull request #15322 from kpedro88/Phase1-HE4

New pulse shape for HE 2017 SiPM

cmsbuild committed:
Merge pull request #15284 from bsunanda/Run2-alca53

bsunanda:Run2-alca53 Change output part for HcalIsolatedBunchFilter

Commit:
Add 2 new condition objects for HCAL: SiPMParameter and SiPMCharacteristics

davidlange6 committed:
Merge pull request #15337 from diguida/fix_HTTP_proxy_curl_EvD_81X

Fix http proxy setting in curl for Tier0 DataService query of the Event Display online application

cmsbuild committed:
Merge pull request #15339 from dntaylor/removeSET

Remove SET muon from constructed

cmsbuild committed:
Merge pull request #15334 from uhussain/fix_staticAnaWarns

Fixed variable names in PFBlockAlgo to abide by CMS coding rules

davidlange6 committed:
Merge pull request #15341 from boudoul/Striphieffi_81X

fixing typo in geometry in Strip hit efficiency tool

davidlange6 committed:
Merge pull request #15345 from fioriNTU/Isolated81X

IsolatedBunchesMonitor81X

davidlange6 committed:
Merge pull request #15320 from kpedro88/HcalRemoveCholeskyObject

remove HcalCholeskyMatrix object entirely

davidlange6 committed:
Merge pull request #15261 from UMN-CMS/umnio

uMNio data unpacking

schneiml committed:
Fix Geometry-Values caching behaviour when step1 EXTEND is used.

This fixes the problem of UNDEFINED values appering in per-lumi trends.

schneiml committed:
Remove the alignment-based geometry code.

It is not really useful anymore, and for future reference it
will be preserved in the history.

Commit:
capitalizing class names

Commit:
oh, lower-cased the xml tag names

ianna committed:
Remove unused varuiable

fioriNTU committed:
Update cfg files for offline analysis

Commit:
Fix a bug for Hcal geometry which affeted Phase2 operation

davidlange6 committed:
Merge pull request #15311 from CTPPS/ctpps_directory_rename

CTPPS: directory rename Totem -> CTPPS

ferencek committed:
added example that makes use of new features in the SV TagInfo producer

VinInn committed:
initialize size as before

VinInn committed:
add errs back as used by JetCoreSpitting

davidlange6 committed:
Merge pull request #15348 from jalimena/UpdateMuonTimingAnalyzer81X

Update muon timing analyzer 81x

Commit:
Fix a spelling

davidlange6 committed:
Merge pull request #15307 from jasperlauwers/81_Mu12_Ele23

81X Adding Mu12_Ele23 path

davidlange6 committed:
Merge pull request #15275 from schneiml/phase1pixeldqm-2dprofiles

Phase1 Pixel DQM Plot Changes

davidlange6 committed:
Merge pull request #15332 from venturia/80x_new_commonmodeanalyzer_sistriptools

New macro for occupancy trend plots and new analyzer for common mode studies

davidlange6 committed:
Merge pull request #15175 from mandrenguyen/pPbBoostVtxSmearing81X

Point to 50ns vtx smearing for boosted pPb and clean up

slava77 commented on CondFormats/HcalObjects/src/HcalInterpolatedTableFunctor.cc:
```diff
@@ -0,0 +1,101 @@
+
+#include "CondFormats/HcalObjects/interface/HcalInterpolatedTableFunctor.h"
+
+inline static double interpolateStep(
```
put this in an anonymous namespace or make less randomly reproducible name?
chances of collision are probably low though (also, one will need to compile with -O0 to make it a visible object)

igv4321 replied:
Hm... I had an impression that static functions
always have internal linkage. Where do I read that
it becomes visible if one compiles with -O0 ?

On 08/04/2016 04:42 PM, Slava Krutelyov wrote:
>
> In CondFormats/HcalObjects/src/HcalInterpolatedTableFunctor.cc <https://github.com/cms-sw/cmssw/pull/15324#discussion_r73608291>:
>
> > @@ -0,0 +1,101 @@
> > +#include <algorithm>
> > +#include "FWCore/Utilities/interface/Exception.h"
> > +
> > +#include "CondFormats/HcalObjects/interface/HcalInterpolatedTableFunctor.h"
> > +
> > +inline static double interpolateStep(
>
> put this in an anonymous namespace or make less randomly reproducible name?
> chances of collision are probably low though (also, one will need to compile with -O0 to make it a visible object)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/cms-sw/cmssw/pull/15324/files/f2734c7b29b226480f1efc887de88c8ae73f121f#r73608291>, or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AFS4-bFAdXXWJEhT7KvCbj2vJxTf36pnks5qclzBgaJpZM4JXio9>.
>

slava77 replied:
recompiled with -O0:
```
nm ../lib/slc6_amd64_gcc530/libCondFormatsHcalObjects.so | grep interpolateStep | c++filt 
000000000027aaea t interpolateStep(double, double, double, double, double)

```

igv4321 replied:
Hm... this behavior is against the C++ standard. Are you sure
that you are not using some special switch, like -fkeep-inline-functions
or -fkeep-inline-functions?

On 08/04/2016 05:42 PM, Slava Krutelyov wrote:
>
> In CondFormats/HcalObjects/src/HcalInterpolatedTableFunctor.cc <https://github.com/cms-sw/cmssw/pull/15324#discussion_r73615584>:
>
> > @@ -0,0 +1,101 @@
> > +#include <algorithm>
> > +#include "FWCore/Utilities/interface/Exception.h"
> > +
> > +#include "CondFormats/HcalObjects/interface/HcalInterpolatedTableFunctor.h"
> > +
> > +inline static double interpolateStep(
>
> recompiled with -O0:
>
> |nm ../lib/slc6_amd64_gcc530/libCondFormatsHcalObjects.so | grep interpolateStep | c++filt 000000000027aaea t interpolateStep(double, double, double, double, double) |
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/cms-sw/cmssw/pull/15324/files/f2734c7b29b226480f1efc887de88c8ae73f121f#r73615584>, or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AFS4-bmxgN0GKFLVCv17pUZJvS9FWvHGks5qcmrEgaJpZM4JXio9>.
>

igv4321 replied:
Sorry, I meant |-fkeep-static-functions for the second switch example
|


On 08/04/2016 05:55 PM, Igor Volobouev wrote:
> Hm... this behavior is against the C++ standard. Are you sure
> that you are not using some special switch, like -fkeep-inline-functions
> or -fkeep-inline-functions?
>
> On 08/04/2016 05:42 PM, Slava Krutelyov wrote:
>>
>> In CondFormats/HcalObjects/src/HcalInterpolatedTableFunctor.cc <https://github.com/cms-sw/cmssw/pull/15324#discussion_r73615584>:
>>
>> > @@ -0,0 +1,101 @@
>> > +#include <algorithm>
>> > +#include "FWCore/Utilities/interface/Exception.h"
>> > +
>> > +#include "CondFormats/HcalObjects/interface/HcalInterpolatedTableFunctor.h"
>> > +
>> > +inline static double interpolateStep(
>>
>> recompiled with -O0:
>>
>> |nm ../lib/slc6_amd64_gcc530/libCondFormatsHcalObjects.so | grep interpolateStep | c++filt 000000000027aaea t interpolateStep(double, double, double, double, double) |
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub <https://github.com/cms-sw/cmssw/pull/15324/files/f2734c7b29b226480f1efc887de88c8ae73f121f#r73615584>, or mute the thread 
>> <https://github.com/notifications/unsubscribe-auth/AFS4-bmxgN0GKFLVCv17pUZJvS9FWvHGks5qcmrEgaJpZM4JXio9>.
>>
>

igv4321 replied:
Does it still show up if you use "nm -g" instead of just "nm"?


On 08/04/2016 05:42 PM, Slava Krutelyov wrote:
>
> In CondFormats/HcalObjects/src/HcalInterpolatedTableFunctor.cc <https://github.com/cms-sw/cmssw/pull/15324#discussion_r73615584>:
>
> > @@ -0,0 +1,101 @@
> > +#include <algorithm>
> > +#include "FWCore/Utilities/interface/Exception.h"
> > +
> > +#include "CondFormats/HcalObjects/interface/HcalInterpolatedTableFunctor.h"
> > +
> > +inline static double interpolateStep(
>
> recompiled with -O0:
>
> |nm ../lib/slc6_amd64_gcc530/libCondFormatsHcalObjects.so | grep interpolateStep | c++filt 000000000027aaea t interpolateStep(double, double, double, double, double) |
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/cms-sw/cmssw/pull/15324/files/f2734c7b29b226480f1efc887de88c8ae73f121f#r73615584>, or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AFS4-bmxgN0GKFLVCv17pUZJvS9FWvHGks5qcmrEgaJpZM4JXio9>.
>

slava77 replied:
``` If lowercase, the symbol is local ``` 
OK, case closed, apparently.

slava77 commented on CondFormats/HcalObjects/src/HcalPiecewiseLinearFunctor.cc:
```diff
@@ -0,0 +1,122 @@
+
+#include "CondFormats/HcalObjects/interface/HcalPiecewiseLinearFunctor.h"
+
+inline static double interpolateSimple(
```
as for interpolateStep

slava77 commented on DataFormats/HcalRecHit/interface/HFPreRecHit.h:
```diff
@@ -28,6 +30,29 @@ class HFPreRecHit
 
+    // Quantities simply added from both anodes
+    float charge() const;
+    float energy() const;
```
will this remain "simply added" ?
Initially, PreRecHit looked like a slightly wrapped access to 1 or 2 HFQIE10Info.
Then, all proper combinations into a single observable would be in HFRecHit.

Maybe make it chargeTotal and energyTotal? 

.. it's minimal enough for now to stay as is, but it seems to me like new functionality is creeping in

igv4321 replied:
This sum is needed in more than one place, so it is best to have
it in this class -- just the basic OO principles.

On 08/04/2016 04:57 PM, Slava Krutelyov wrote:
>
> In DataFormats/HcalRecHit/interface/HFPreRecHit.h <https://github.com/cms-sw/cmssw/pull/15324#discussion_r73610270>:
>
> > @@ -28,6 +30,29 @@ class HFPreRecHit
> >      // index is out of range.
> >      const HFQIE10Info* getHFQIE10Info(unsigned index) const;
> >
> > +    // Quantities simply added from both anodes
> > +    float charge() const;
> > +    float energy() const;
>
> will this remain "simply added" ?
> Initially, PreRecHit looked like a slightly wrapped access to 1 or 2 HFQIE10Info.
> Then, all proper combinations into a single observable would be in HFRecHit.
>
> Maybe make it chargeTotal and energyTotal?
>
> .. it's minimal enough for now to stay as is, but it seems to me like new functionality is creeping in
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/cms-sw/cmssw/pull/15324/files/f2734c7b29b226480f1efc887de88c8ae73f121f#r73610270>, or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AFS4-Q4SOATYQRW4Uil7WMzuLPxs8fRmks5qcmBGgaJpZM4JXio9>.
>

slava77 commented on DataFormats/METReco/interface/HcalPhase1FlagLabels.h:
```diff
@@ -0,0 +1,32 @@
+        /* 11 */ HBHEIsolatedNoise = HcalCaloFlagLabels::HBHEIsolatedNoise,
+        /* 12 */ HBHEFlatNoise = HcalCaloFlagLabels::HBHEFlatNoise,
+        /* 13 */ HBHESpikeNoise = HcalCaloFlagLabels::HBHESpikeNoise,
+        /* 15 */ HBHETS4TS5Noise = HcalCaloFlagLabels::HBHETS4TS5Noise,
```
why not simply edit HcalCaloFlagLabels?
partial reuse of the same values looks odd.

IMO, These flags are too high level to pass downstream support of different meanings for phase1 vs not phase1.
We will need to support re-reco of "legacy" HF. 
It seems simpler to preserve common phase space for flag enums.

igv4321 replied:
It is good to maintain proper history of flag settings. We intend
to change some bit settings, not just add the new ones.

On 08/04/2016 05:17 PM, Slava Krutelyov wrote:
>
> In DataFormats/METReco/interface/HcalPhase1FlagLabels.h <https://github.com/cms-sw/cmssw/pull/15324#discussion_r73612580>:
>
> > @@ -0,0 +1,32 @@
> > +#ifndef DataFormats_METReco_HcalPhase1FlagLabels_h
> > +#define DataFormats_METReco_HcalPhase1FlagLabels_h
> > +
> > +#include "DataFormats/METReco/interface/HcalCaloFlagLabels.h"
> > +
> > +namespace HcalPhase1FlagLabels
> > +{
> > +    enum HBHEStatusFlag {
> > +        /*  0 */ HBHEHpdHitMultiplicity = HcalCaloFlagLabels::HBHEHpdHitMultiplicity,
> > +        /* 11 */ HBHEIsolatedNoise = HcalCaloFlagLabels::HBHEIsolatedNoise,
> > +        /* 12 */ HBHEFlatNoise = HcalCaloFlagLabels::HBHEFlatNoise,
> > +        /* 13 */ HBHESpikeNoise = HcalCaloFlagLabels::HBHESpikeNoise,
> > +        /* 15 */ HBHETS4TS5Noise = HcalCaloFlagLabels::HBHETS4TS5Noise,
>
> why not simply edit HcalCaloFlagLabels?
> partial reuse of the same values looks odd.
>
> IMO, These flags are too high level to pass downstream support of different meanings for phase1 vs not phase1.
> We will need to support re-reco of "legacy" HF.
> It seems simpler to preserve common phase space for flag enums.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/cms-sw/cmssw/pull/15324/files/f2734c7b29b226480f1efc887de88c8ae73f121f#r73612580>, or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AFS4-XtJAMxoTjJTi7ZGZ5hjIRcAfyywks5qcmTegaJpZM4JXio9>.
>

slava77 replied:
if you plan to change meaning/naming of the same enum values, what is your plan to support 2016 HCAL ?
If both 2016 and future phase1 will be handled by HcalPhase1FlagLabels, it's probably OK
(HcalCaloFlagLabels will just become deprecated and eventually go away).

Please clarify.

igv4321 replied:
We will see what happens... There is no decision yet on how 2016
HCAL will be handled. For now, both options are open -- we can
switch or we can maintain the legacy code.

On 08/04/2016 05:47 PM, Slava Krutelyov wrote:
>
> In DataFormats/METReco/interface/HcalPhase1FlagLabels.h <https://github.com/cms-sw/cmssw/pull/15324#discussion_r73616084>:
>
> > @@ -0,0 +1,32 @@
> > +#ifndef DataFormats_METReco_HcalPhase1FlagLabels_h
> > +#define DataFormats_METReco_HcalPhase1FlagLabels_h
> > +
> > +#include "DataFormats/METReco/interface/HcalCaloFlagLabels.h"
> > +
> > +namespace HcalPhase1FlagLabels
> > +{
> > +    enum HBHEStatusFlag {
> > +        /*  0 */ HBHEHpdHitMultiplicity = HcalCaloFlagLabels::HBHEHpdHitMultiplicity,
> > +        /* 11 */ HBHEIsolatedNoise = HcalCaloFlagLabels::HBHEIsolatedNoise,
> > +        /* 12 */ HBHEFlatNoise = HcalCaloFlagLabels::HBHEFlatNoise,
> > +        /* 13 */ HBHESpikeNoise = HcalCaloFlagLabels::HBHESpikeNoise,
> > +        /* 15 */ HBHETS4TS5Noise = HcalCaloFlagLabels::HBHETS4TS5Noise,
>
> if you plan to change meaning/naming of the same enum values, what is your plan to support 2016 HCAL ?
> If both 2016 and future phase1 will be handled by HcalPhase1FlagLabels, it's probably OK
> (HcalCaloFlagLabels will just become deprecated and eventually go away).
>
> Please clarify.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/cms-sw/cmssw/pull/15324/files/f2734c7b29b226480f1efc887de88c8ae73f121f#r73616084>, or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AFS4-Sz8C7nvIJGSchAD0Ih6qEMHhtG5ks5qcmvwgaJpZM4JXio9>.
>

slava77 replied:
was this flag duality clearly agreed in HCAL DPG and JME POG? (JME is the primary customer of the flags). Now it looks like you are creating a parallel universe of flags for the phase-1 hits.

@mariadalfonso @mmarionncern @schoef

igv4321 replied:
The flags were discussed with the Hcal Noise Group.
All flags that are currently being used for noise cleaning
are preserved.

I always had an impression that JME is just using
HcalNoiseSummary, and the code making that
object is maintained by the Hcal Noise Group.

On 08/04/2016 07:48 PM, Slava Krutelyov wrote:
>
> In DataFormats/METReco/interface/HcalPhase1FlagLabels.h <https://github.com/cms-sw/cmssw/pull/15324#discussion_r73627401>:
>
> > @@ -0,0 +1,32 @@
> > +#ifndef DataFormats_METReco_HcalPhase1FlagLabels_h
> > +#define DataFormats_METReco_HcalPhase1FlagLabels_h
> > +
> > +#include "DataFormats/METReco/interface/HcalCaloFlagLabels.h"
> > +
> > +namespace HcalPhase1FlagLabels
> > +{
> > +    enum HBHEStatusFlag {
> > +        /*  0 */ HBHEHpdHitMultiplicity = HcalCaloFlagLabels::HBHEHpdHitMultiplicity,
> > +        /* 11 */ HBHEIsolatedNoise = HcalCaloFlagLabels::HBHEIsolatedNoise,
> > +        /* 12 */ HBHEFlatNoise = HcalCaloFlagLabels::HBHEFlatNoise,
> > +        /* 13 */ HBHESpikeNoise = HcalCaloFlagLabels::HBHESpikeNoise,
> > +        /* 15 */ HBHETS4TS5Noise = HcalCaloFlagLabels::HBHETS4TS5Noise,
>
> was this flag duality clearly agreed in HCAL DPG and JME POG? (JME is the primary customer of the flags). Now it looks like you are creating a parallel universe of flags for the phase-1 hits.
>
> @mariadalfonso <https://github.com/mariadalfonso> @mmarionncern <https://github.com/mmarionncern> @schoef <https://github.com/schoef>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/cms-sw/cmssw/pull/15324/files/f2734c7b29b226480f1efc887de88c8ae73f121f#r73627401>, or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AFS4-SO0ydDyCp9GLnbGlsWw6vjq_3i3ks5qcoh0gaJpZM4JXio9>.
>

ianna committed:
Fix discrepancy between G4 and Root parameters

slava77 commented:
The non-Hcal-specific generic code, which is a bulk of CondFormats/HcalObjects looks somewhat out of place.
Something for the alca and db to decide on.

davidlange6 committed:
Merge pull request #15361 from fioriNTU/UpdateOccupancyCFG

Update cfg files for offline analysis

davidlange6 committed:
remove failing relval

igv4321 commented:
I am afraid that, following your logic, one will have to decide that
Hcal itself is not Hcal-specific. After all, it is built out of generic
copper, steel, and scintillator.

On 08/05/2016 12:51 AM, Slava Krutelyov wrote:
>
> The non-Hcal-specific generic code, which is a bulk of CondFormats/HcalObjects looks somewhat out of place.
> Something for the alca and db to decide on.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/cms-sw/cmssw/pull/15324#issuecomment-237761932>, or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AFS4-TqXBeWNwxeRt6pJS-owUM-UVFfxks5qcs9ngaJpZM4JXio9>.
>

davidlange6 committed:
Merge pull request #15370 from davidlange6/noHIFail

remove failing relval from short matrix (temporarily)

davidlange6 committed:
Merge pull request #15347 from ianna/dd-cuttubs

Cylindrical Cut Section or Cut Tube Implementation

abdoulline commented:
I presume Slava meant interpolation/polynomials etc. algebra can be put 
on a "neutral territory", like CalibCalorimetry/HcalAlgos?


On Fri, 5 Aug 2016, Igor Volobouev wrote:

> I am afraid that, following your logic, one will have to decide that
> Hcal itself is not Hcal-specific. After all, it is built out of generic
> copper, steel, and scintillator.
> 
> On 08/05/2016 12:51 AM, Slava Krutelyov wrote:
> >
> > The non-Hcal-specific generic code, which is a bulk of CondFormats/HcalObjects
> looks somewhat out of place.
> > Something for the alca and db to decide on.
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/15324#issuecomment-237761932>, or mute the
> thread
> ><https://github.com/notifications/unsubscribe-auth/AFS4-TqXBeWNwxeRt6pJS-owUM-UVF
> fxks5qcs9ngaJpZM4JXio9>.
> >
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the
> thread.[AEx02n5fVlMr2l1gbN9vAUJMGhAx18c9ks5qcvkGgaJpZM4JXio9.gif]
> 
> 
>

davidlange6 commented:
Actually its more a question of how many mini-generic-infrastructures you want in detector specific areas. The only reason much of the HcalObjects code is not generic and thus easily used by others is the name of the class. Its a good way to save effort for CMS…
Anyway, @slava77 correctly suggests its for alca/db to discuss.

On Aug 5, 2016, at 11:34 AM, Salavat Abdullin <notifications@github.com<mailto:notifications@github.com>> wrote:


I presume Slava meant interpolation/polynomials etc. algebra can be put
on a "neutral territory", like CalibCalorimetry/HcalAlgos?


On Fri, 5 Aug 2016, Igor Volobouev wrote:

> I am afraid that, following your logic, one will have to decide that
> Hcal itself is not Hcal-specific. After all, it is built out of generic
> copper, steel, and scintillator.
>
> On 08/05/2016 12:51 AM, Slava Krutelyov wrote:
> >
> > The non-Hcal-specific generic code, which is a bulk of CondFormats/HcalObjects
> looks somewhat out of place.
> > Something for the alca and db to decide on.
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/15324#issuecomment-237761932>, or mute the
> thread
> ><https://github.com/notifications/unsubscribe-auth/AFS4-TqXBeWNwxeRt6pJS-owUM-UVF
> fxks5qcs9ngaJpZM4JXio9>.
> >
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the
> thread.[AEx02n5fVlMr2l1gbN9vAUJMGhAx18c9ks5qcvkGgaJpZM4JXio9.gif]
>
>
>

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/cms-sw/cmssw/pull/15324#issuecomment-237806623>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AEzywxsU_KpEzqbHmAi3aIK3yBHGlO2Lks5qcwOKgaJpZM4JXio9>.

igv4321 commented:
Fine. Please make sure that the timing of this discussion
does not delay the Hcal software development effort by
another month or so.

On 08/05/2016 04:43 AM, David Lange wrote:
> Actually its more a question of how many mini-generic-infrastructures you want in detector specific areas. The only reason much of the HcalObjects code is not generic and thus easily used by others 
> is the name of the class. Its a good way to save effort for CMS…
> Anyway, @slava77 correctly suggests its for alca/db to discuss.
>
> On Aug 5, 2016, at 11:34 AM, Salavat Abdullin <notifications@github.com<mailto:notifications@github.com>> wrote:
>
>
> I presume Slava meant interpolation/polynomials etc. algebra can be put
> on a "neutral territory", like CalibCalorimetry/HcalAlgos?
>
>
> On Fri, 5 Aug 2016, Igor Volobouev wrote:
>
> > I am afraid that, following your logic, one will have to decide that
> > Hcal itself is not Hcal-specific. After all, it is built out of generic
> > copper, steel, and scintillator.
> >
> > On 08/05/2016 12:51 AM, Slava Krutelyov wrote:
> > >
> > > The non-Hcal-specific generic code, which is a bulk of CondFormats/HcalObjects
> > looks somewhat out of place.
> > > Something for the alca and db to decide on.
> > >
> > > —
> > > You are receiving this because you were mentioned.
> > > Reply to this email directly, view it on GitHub
> > <https://github.com/cms-sw/cmssw/pull/15324#issuecomment-237761932>, or mute the
> > thread
> > ><https://github.com/notifications/unsubscribe-auth/AFS4-TqXBeWNwxeRt6pJS-owUM-UVF
> > fxks5qcs9ngaJpZM4JXio9>.
> > >
> >
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub, or mute the
> > thread.[AEx02n5fVlMr2l1gbN9vAUJMGhAx18c9ks5qcvkGgaJpZM4JXio9.gif]
> >
> >
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub<https://github.com/cms-sw/cmssw/pull/15324#issuecomment-237806623>, or mute the 
> thread<https://github.com/notifications/unsubscribe-auth/AEzywxsU_KpEzqbHmAi3aIK3yBHGlO2Lks5qcwOKgaJpZM4JXio9>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/cms-sw/cmssw/pull/15324#issuecomment-237808589>, or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AFS4-TJg2jWKQBbAkmCjyfQCK6CTyhW0ks5qcwXNgaJpZM4JXio9>.
>

davidlange6 committed:
Merge pull request #15352 from bsunanda/Run2-alca55

bsunanda:Run2-alca55 Correct G4 validation code using isolated particles

davidlange6 committed:
Merge pull request #14998 from gartung/PhysicsTools-MVATrainer-fix-indent-not-guard

PhysicsTools/MVATrainer : formatting fix for gcc 6.0 misleading-indentation warning

davidlange6 committed:
Merge pull request #15034 from gartung/DQM-SiPixelMonitorClient-fix-indent-not-guard

DQM/SiPixelMonitorClient : formatting fix for gcc 6.0 misleading-indentation warning

davidlange6 committed:
Merge pull request #15072 from gartung/SimCalorimetry-EcalTrigPrimProducers-fix-indent-not-guard-warn

SimCalorimetry/EcalTrigPrimProducers fix for bug flagged by gcc 6.0 misleading-indentation warning

Commit:
Complete the analyzer and correct the configuration file for production

Commit:
Update all codes for HGCal TB till July 2016

mmusich committed:
fixing logic to trigger update in SiPixelAli wf

slava77 commented:
for https://github.com/cms-sw/cmssw/pull/15324 f2734c7
- code changes are in line with the description (more detailed in the slides)
- jenkins tests pass; the modified code is not a part of standard workflows
- tested locally based on provided instructions using the payload from DB (actually, the test used TTbar PU35 GEN-SIMed and digitized in 810pre8 and re-reconstructed as prescribed in pre9+#15305). Comparisons with baseline show changes only in HFRecHits flags (flag at bit 20 = TimingFromTDC is now always set); this is in agreement with the visual code inspection (since a large fraction of logic is encoded in the calibration DB, I'm just assuming that the DB values are somewhat simple; I did not inspect the payloads).

@igv4321 please confirm that this behavior is expected (only flags change: from 0 to only bit20 set).

cmsbuild committed:
Merge pull request #15359 from makortel/fixSeedingLayerSetsHits

Two technical fixes to SeedingLayerSetsHits

cmsbuild committed:
Merge pull request #15318 from VinInn/PixelRedux

Update pixelCluster to support PhaseII

davidlange6 committed:
Merge pull request #14963 from dan131riley/HcalTasks-rawtask-bad-event-protection

protect RawTask::_process from processing bad HcalHTRData

igv4321 commented:
Yes, this looks fine. The payload was, basically, "all pass",
so that no TDC timing measurements would be rejected
and no charge asymmetry values would be flagged.

In principle, the code also sets a number of other flags,
but since we do not model detector noise in MC, those
are difficult to check.

On 08/05/2016 11:03 PM, Slava Krutelyov wrote:
>
> for #15324 <https://github.com/cms-sw/cmssw/pull/15324> f2734c7 <https://github.com/cms-sw/cmssw/commit/f2734c7b29b226480f1efc887de88c8ae73f121f>
>
>   * code changes are in line with the description (more detailed in the slides)
>   * jenkins tests pass; the modified code is not a part of standard workflows
>   * tested locally based on provided instructions using the payload from DB (actually, the test used TTbar PU35 GEN-SIMed and digitized in 810pre8 and re-reconstructed as prescribed in pre9+#15305
>     <https://github.com/cms-sw/cmssw/pull/15305>). Comparisons with baseline show changes only in HFRecHits flags (flag at bit 20 = TimingFromTDC is now always set); this is in agreement with the
>     visual code inspection (since a large fraction of logic is encoded in the calibration DB, I'm just assuming that the DB values are somewhat simple; I did not inspect the payloads).
>
> @igv4321 <https://github.com/igv4321> please confirm that this behavior is expected (only flags change: from 0 to only bit20 set).
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/cms-sw/cmssw/pull/15324#issuecomment-238004410>, or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AFS4-fWNybbyP_njs7YIrNSNGq1hXCwqks5qdAeigaJpZM4JXio9>.
>

slava77 commented:
+1

for https://github.com/cms-sw/cmssw/pull/15305 86f675d

details in https://github.com/cms-sw/cmssw/pull/15324#issuecomment-238004410

davidlange6 committed:
Merge pull request #15363 from kpedro88/HcalRemoveCovarianceObject

remove HcalCovarianceMatrix object entirely

cmsbuild committed:
Merge pull request #15379 from bsunanda/Run2-TB04

bsunanda:Run2-TB04 Update all codes for HGCal TB till July 2016

cmsbuild committed:
Merge pull request #15377 from bsunanda/Run2-TB03

bsunanda:Run2-TB03 Complete the analyzer and correct the configuration file for production

cmsbuild committed:
Merge pull request #15237 from gouskos/pix-endcap-roc-81x

dqm plot, pixel endcap roc occupancy in 81x

Andrej-CMS committed:
fixed '.' in config

cmsbuild committed:
Merge pull request #15301 from Andrej-CMS/ttHFGenFilter

TTHFGenFilter

mmusich committed:
make sure updateDB is false when starting reading res file

uhussain committed:
arrays nyzero and ysum initialized to zero without using the for loop in line 220. The for loop is unreliable because the bound of the loop is an input argument that need not match the size of the array

franzoni commented:
dear @igv4321 @abdoulline 
we feel that moving everything in this PR which is not a strictly in condformat
would keep the packages definition
and the distinction between AlCa and Db signatures
more clear.
Would you mind helping with that ?

Thanks, Marco @mmusich and Giovanni

ferencek committed:
improved handling of wrong input jet types

uhussain committed:
for loop is back up with the change

uhussain committed:
initialize outside the for loop

mmusich committed:
break loop on alignables if exceeds maximum observed movements or erros

igv4321 commented:
"...moving everything in this PR which is not a strictly in condformat...".

Please clarify what is "strictly condformat" and "not strictly condformat",
and where you propose to move the code. I am sure we will be able to
help once we understand what is that you want to achieve.

On 08/08/2016 07:27 AM, Giovanni Franzoni wrote:
>
> dear @igv4321 <https://github.com/igv4321> @abdoulline <https://github.com/abdoulline>
> we feel that moving everything in this PR which is not a strictly in condformat
> would keep the packages definition
> and the distinction between AlCa and Db signatures
> more clear.
> Would you mind helping with that ?
>
> Thanks, Marco @mmusich <https://github.com/mmusich> and Giovanni
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/cms-sw/cmssw/pull/15324#issuecomment-238220674>, or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AFS4-bF-CUfvxQgfwrGvUF2-sKl_4-7bks5qdyCvgaJpZM4JXio9>.
>

mmusich committed:
streamline code, remove un-necessary members

mmusich committed:
Revert "streamline code, remove un-necessary members"

This reverts commit 95e72d226b7812f94bd1f284887e91a4d9f8e261.

davidlange6 committed:
Merge pull request #15371 from makortel/trackTPMatchingPixelWeight

Add possibility to weigh pixel hits in QuickTrackAssociatorByHits

davidlange6 committed:
Merge pull request #15325 from dildick/from-CMSSW_8_1_0_pre9-stop-produce-gem-copad-digi-2016

Don't produce GEMCoPadDigi in 2016 WFs

cmsbuild committed:
Merge pull request #15386 from uhussain/fix_staticAnaWarns

Replacing unreliable array initialization in SiPixelTemplateReco.cc

mmusich committed:
Removing unused private members

davidlange6 committed:
Merge pull request #15331 from cms-btv-pog/SubjetTagInfosNoClustering-PR_from-CMSSW_8_1_0_pre9

Subjet TagInfo updates

cmsbuild committed:
Merge pull request #15385 from mmusich/81X_fixSiPixelAliPCL

81x: fix SiPixelAli PCL update logic

davidlange6 committed:
Merge pull request #15243 from kkotov/extendedO2Ofor81X

Extended o2o for81x

davidlange6 committed:
Merge pull request #15065 from gartung/GeneratorInterface-LHEInterface-fix-indent-not-guard

GeneratorInterface/LHEInterface : formatting fix for statements flagged by gcc 6.0 indentation warning

cmsbuild committed:
Merge pull request #15364 from bsunanda/Run2-hcx87

Run2-hcx87 Fix a bug for Hcal geometry which affeted Phase2 operation

davidlange6 committed:
Merge pull request #15357 from bsunanda/Run2-alca57

bsunanda:Run2-alca57 Add 2 new condition objects for HCAL: SiPMParameter and SiPMCharacteristics

kpedro88 commented:
@igv4321 needs to be rebased, some merge conflict has appeared

Commit:
Merged realistic-phase1-hf-reco from repository igv4321

Conflicts:
	CondCore/HcalPlugins/src/plugin.cc

cmsbuild commented:
Pull request #15324 was updated. @ghellwig, @ianna, @cerminar, @fabozzi, @Martin-Grunewald, @rekovic, @vanbesien, @perrotta, @civanch, @franzoni, @cmsbuild, @fwyzard, @davidlange6, @Dr15Jones, @cvuosalo, @mdhildreth, @slava77, @ggovi, @mmusich, @mulhearn, @dmitrijus, @srimanob, @hengne can you please check and sign again.

kpedro88 commented:
@igv4321 that rebase did not work.

igv4321 commented:
The conflict should be fixed now

kpedro88 commented:
@igv4321 you somehow pulled in 250+ previous commits. Did you rebase or merge?

igv4321 commented:
Yes, this became a mess... Trying to revert now

igv4321 commented:
Fixing this cleanly is beyond my understanding of Git. Submitting PR #15419 instead.
